src/QsCompiler/CompilationManager/CompilationUnit.cs

```diff
@@ -1043,16 +1052,15 @@ internal LocalDeclarations TryGetLocalDeclarations(FileContentManager file, Posi
             {
                 throw new ArgumentNullException(nameof(types));
             }
-            predicate ??= _ => true;
 
             // Assign a unique ID to each reference.
 
-            var decorator = new NameDecorator($"QsRef");
-            var ids =
+            NameDecorator decorator = new NameDecorator($"QsRef");
+            ImmutableDictionary<string, int> ids =
```

Comment:
Can these be `var`?

Reply:
Unfortunately, no, because both of them are inferred as nullable types for some reason, which causes an error on line 1078 when `GetMappingForSourceGroup` tries to use them. I'm not sure why this happens.